pkg/data/generic.go

```diff
@@ -203,31 +196,81 @@ func (s *GenericSync) sync(resource dynamic.NamespaceableResourceInterface, quit
 }
 
 func (s *GenericSync) syncAdd(obj runtime.Object) error {
-	m, err := meta.Accessor(obj)
+	path, err := s.objPath(obj)
 	if err != nil {
 		return err
 	}
-	name := m.GetName()
-	var path = m.GetName()
-	if s.ns.Namespaced {
-		path = m.GetNamespace() + "/" + name
-	}
 	return s.opa.PutData(path, obj)
 }
 
 func (s *GenericSync) syncRemove(obj runtime.Object) error {
-	m, err := meta.Accessor(obj)
+	path, err := s.objPath(obj)
+	if err != nil {
+		return err
+	}
+	return s.opa.PatchData(path, opa_client.DataPatchOpRemove, nil)
+}
+
+func (s *GenericSync) syncAll(objs []unstructured.Unstructured) error {
+
+	// Build a list of patches to apply.
+	var patches []opa_client.DataPatch
+
+	createdSubPaths := map[string]struct{}{}
+
+	for _, obj := range objs {
```

Comment:
I think it would be worthwhile to factor this into a separate function and then test for a set of objects that the desired patch set is generated.

Reply:
Ah yea, sorry, should have probably marked this as WIP. Still need to add a bunch of test cases. Didn't want to go too deep on them until getting some feedback on direction.

I like separating it out though, will make it easier to write unit tests for 👍

Reply:
Latest updates break it up and add tests for generating patches.